examples/with-react-intl/components/PageWithIntl.js

```diff
@@ -0,0 +1,44 @@
+import React, {Component} from 'react'
+import {IntlProvider, addLocaleData, injectIntl} from 'react-intl'
+
+// Register React Intl's locale data for the user's locale in the browser. This
+// locale data was added to the page by `pages/_document.js`. This only happens
+// once, on initial page load in the browser.
+if (typeof window !== 'undefined' && window.ReactIntlLocaleData) {
```

Comment:
Thanks for the details 🙂 

1. IMHO worth to create a bug on Next.js
2. the localeDataScript itself is just a string, which is serializable. Then the string could be evaluated where needed in the HOC (e.g with `eval`)? Right now the intl code is spread all around and that's hardly maintainable.
3. should work by returning a `serverRendered: !process.browser` in your getInitialProps and wrap the `addLocaleData` in `if (props.serverRendered)`?: https://github.com/relatenow/relate/blob/ee51f1dd6990177ff0924b1c0d78fddfd23aa564/hocs/withIntl.js#L43